### PR TITLE
Bump typescript, @typescript-eslint/eslint-plugin, and @typescript-eslint/parser

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

These three dependencies need to be bumped together or else building and linting fails.

### CODE REVIEW

File: src/common/k8s-api/endpoints/metrics.api.ts [hunk 1]
```diff
@@ -55,28 +54,33 @@ export interface IResourceMetrics<T extends MetricData> {
...
+async function getMetrics(query: string | string[] | Partial<Record<string, Partial<Record<string, string>>>>, reqParams: IMetricsReqParams = {}): Promise<MetricData | MetricData[] | Partial<Record<string, MetricData>>> {
+  const { range = 3600, step = 60, namespace } = reqParams;
+  let { start, end } = reqParams;
 
-      start = now - range;
-      end = now;
-    }
+  if (!start && !end) {
+    const timeNow = Date.now() / 1000;
+    const now = moment.unix(timeNow).startOf("minute").unix();  // round date to minutes
```
Comment (aleksfront) [inline src/common/k8s-api/endpoints/metrics.api.ts:67]: Maybe cleaner to move metrics and router part to separate PR. Wdyt?

Comment (Nokel81) [reply src/common/k8s-api/endpoints/metrics.api.ts:67]: I had to do this to fix the type error because you cannot specify overrides in object notation.

**Review by Iku-turso — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]
